proposals/explicit-backing-fields.md

```diff
@@ -14,417 +14,396 @@ but one is part of a public API and another is an implementation detail.
 
-In some application it is desirable to expose not just a read-only subtype, but a specially constructed read-only
-view that protects the data structure from casting into mutable type. For example a `MutableStateFlow` has
-[`asStateFlow`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/as-state-flow.html)
-extension for that purpose.
+#### Hide complex value storage logic
 
-With the proposed syntax it is possible to declare a custom getter:
+Sometimes we want to provide a public API for getting the instant value of a variable,
+and hide complex storage logic inside the implementation.
+For example, we may store `AtomicInt` and provide api for obtaining instant `Int` value.
 
 ```kotlin
-val state: StateFlow<State> 
-    get() = field.asStateFlow()
-    field = MutableStateFlow(State.INITIAL)
+private val _itemCount = atomic(0)
+val itemCount: Int get() = _itemCount.value
 ```
 
-> Note, that `field.asStateFlow()` allocates a wrapper read-only view object and it is called here every time this 
-property is accessed, allocating a new instance. For some applications this is an acceptable and even preferable 
-behavior. For others, it is crucial to create this view object once and cache it in a separate field. We don't plan 
-to address the later use-cases &mdash; that's where the backing-field pattern will have to continue to be used.
+It is also possible with other containers:
+
+- Other atomic classes ([example](https://github.com/Sorapointa/Sorapointa/blob/31578cb9e460bb4516b51fcef4f0c4af5600caa9/sorapointa-core/src/main/kotlin/org/sorapointa/game/PlayerAvatarComp.kt#L42-L82))
+- `StateFlow`, `LiveData`, etc ([example](https://github.com/c5inco/Compose-Modifiers-Playground/blob/2c7e8c55eeaab15e38696397a84b57475a84a6a3/ideaPlugin/src/main/kotlin/intellij/SwingColors.kt#L44-L52))
+- `ThreadLocal` ([example](https://github.com/corda/corda/blob/a95b854b1eda8a1e9284c0db8ff43b78ea0eb290/core/src/main/kotlin/net/corda/core/serialization/SerializationAPI.kt#L56))
+- `WeakReference` ([example](https://github.com/expo/expo/blob/a502f90d6d9345648157aceb16a78f044669842e/packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/devtools/DevMenuDevToolsDelegate.kt#L21-L32))
 
-For this use-case to become actually usable, the TBD syntax of [Direct Backing Field Access](#direct-backing-field-access) 
-will need to be added. That's because such applications need to modify the mutable data structure that is stored in the field,
-but the rules of [Smart Type Narrowing](#smart-type-narrowing) will not make it directly available. 
+#### Access backing property inside
 
-### Access field from outside of getter and setter
+It is worth emphasizing that in most cases the primary property is only for public use and is not
+referenced within the class. Obviously, there is no point in doing so, since the backing property usually provides
+all the same functionality, as a primary property and even more.
```

Comment:
Please don't use the word "obviously" in such a context. This is not a conclusion within a set of strict constraints, but the rationale behind a free choice we decide to make